### PR TITLE
AO3-6065 Add task to set rating to "Not Rated" if missing

### PR DESCRIPTION
**Pull request by tlee911**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6065

## Purpose

Adds a rake task `rake After:add_default_rating_to_works` which will find works missing a rating and set them to the canonical "Not Rated" instead.

## Testing Instructions

- Find some works which have no rating (or delete the ratings yourself with DB access)
- `rake After:add_default_rating_to_works`
- Verify the works in question are now tagged "Not Rated"
- Verify works which had valid ratings were not changed

Automated tests have also been added in `after_tasks.rake_spec.rb`

### CODE REVIEW

File: lib/tasks/work_tasks.rake
Comment (sarken) [inline lib/tasks/work_tasks.rake:32]: We could probably avoid tucking everything inside an `if...end` by doing `next unless work.rating_string.blank?` or `next if work.rating_string.present?` instead.

File: lib/tasks/work_tasks.rake
Comment (sarken) [inline lib/tasks/work_tasks.rake:29]: Since this is intended to be a one-time task, it would be more at home in `lib/tasks/after_tasks.rake`.

It also needs some tests, which can be added to `spec/lib/tasks/after_tasks.rake_spec.rb`.

File: lib/tasks/work_tasks.rake
Comment (sarken) [inline lib/tasks/work_tasks.rake:31]: Since this is going to take a while, can we use `find_in_batches` and then print a progress indicator to `STDOUT` so we can see how things are progressing? (There are a bunch of examples of possible approaches in the other recent after tasks.)

File: lib/tasks/work_tasks.rake
Comment (sarken) [inline lib/tasks/work_tasks.rake:28]: We'll be able to check the history or blame if we ever need to know what issue this task was for (as opposed to what it does, which is covered in the description), so this can be left off.

**Review by sarken — CHANGES_REQUESTED** [review]
Thank you for working on this!

**Comment by tlee911** [comment]
I spent all morning on the test code, and I still can't figure out what's wrong with it 😕, so help would be appreciated!

**Comment by tlee911** [comment]
@tickinginstant helped me figure it out.  Turns out not all the ratings are available to tests by default and I happened to pick one (mature) that didn't exist.  I've updated spec_helper.rb to have all the usual ratings so this doesn't surprise anyone else in the future.

**Review by sarken — CHANGES_REQUESTED** [review]
We were just talking about potentially making some more changes to this due to [AO3-5988](https://otwarchive.atlassian.net/browse/AO3-5988), so no rush on anything I've mentioned here!

File: spec/spec_helper.rb
Comment (tlee911) [reply spec/spec_helper.rb:54]: Ok, I can back out this change.  I did end up switching to explicit (was trying to use mature before) which does exist in tests so this is not strictly necessary for my tests to pass.

**Comment by tlee911** [comment]
After looking at #4003, I realize I'm manipulating ratings via `ratings_string` rather than the `ratings` array.  While both seem to work, is there a preference for using `ratings` directly?  That feels more right to me.

**Comment by sarken** [comment]
Manipulating the array directly sounds good to me!

**Review by sarken — APPROVED** [review]
I think that's everything now -- thank you!

**Comment by redsummernight** [comment]
This has conflicts now, but let's wait until #4022 is merged first (since that'll also introduce conflicts).

**Comment by redsummernight** [comment]
This has conflicts now, but we can merge it next. You should also pull from latest master to fix the dependency security warnings.

**Pull request merged**
